File: src/desktop/ui/src/Analysis/Models/ReadingModel.ts
import type { ReadingArea } from "./ReadingArea";
import type { ReadingAssurance } from "./ReadingAssurance";
import type { ReadingCitation } from "./ReadingCitation";
import type { ReadingEvidence } from "./ReadingEvidence";
import type { ReadingLimitation } from "./ReadingLimitation";
import type { ReadingOmissionSummary } from "./ReadingOmissionSummary";
import type { ReadingStatement } from "./ReadingStatement";

export interface ReadingModel {
  readonly thesis: ReadingStatement | null;
  readonly areas: readonly ReadingArea[];
  readonly citations: readonly ReadingCitation[];
  readonly evidence: readonly ReadingEvidence[];
  readonly limitations: readonly ReadingLimitation[];
  readonly omissionSummaries: readonly ReadingOmissionSummary[];
  readonly assurances: readonly ReadingAssurance[];
}

File: src/desktop/src-tauri/src/analysis/models/reading_model.rs
use crate::analysis::models::{
    ReadingArea, ReadingAssurance, ReadingCitation, ReadingEvidence, ReadingLimitation,
    ReadingOmissionSummary, ReadingStatement,
};
use serde::{Deserialize, Serialize};

/// Represents the complete reading model published with an analysis run.
#[derive(Clone, Debug, Deserialize, PartialEq, Eq, Serialize)]
#[serde(rename_all = "camelCase", deny_unknown_fields)]
pub struct ReadingModel {
    pub thesis: Option<ReadingStatement>,
    pub areas: Vec<ReadingArea>,
    pub citations: Vec<ReadingCitation>,
    pub evidence: Vec<ReadingEvidence>,
    pub limitations: Vec<ReadingLimitation>,
    pub omission_summaries: Vec<ReadingOmissionSummary>,
    pub assurances: Vec<ReadingAssurance>,
}
